apps/openmw/mwrender/actoranimation.cpp

```diff
@@ -234,7 +235,11 @@ void ActorAnimation::updateHolsteredShield(bool showCarriedLeft)
     }
 
     if (mAlpha != 1.f)
-        mResourceSystem->getSceneManager()->recreateShaders(mHolsteredShield->getNode());
+    {
+        // TODO: Remove when we have figured out why this was ever necessary.
```

Comment:
A little off-topic, but I figured this out. The tl;dr is that it was never necessary and arose due to a dodgy rebase.

When effects like chameleon are used, actors can become translucent. This is done by changing the material colour, and seems to be done that way in the original engine, too, meaning that vertex colours are ignored for translucent actors. In the olden days, we set the colour mode (whether to use vertex colours of the material colour) via an `@define`, but the original shadows PR changed it to be a uniform as it made implementing something a little easier and looked likely to give better performance. Before that change it was necessary to `recreateShaders` when things first became affected by chameleon, and this used to be the *vast* majority of `recreateShaders` calls (in the code, not at runtime). In early 2019, bzzt noticed we'd left all these `recreateShaders` calls in, when all they really needed to do was change a uniform, so he implemented this MR to switch to just overriding the uniform: https://gitlab.com/OpenMW/openmw/-/commit/5ec28e09c9707f586812bd7d1b1823028f866e07. I'm sure you'll agree that that was a good move.

In 2018, akortunov started working on the shield holstering feature in a branch (this PR https://github.com/OpenMW/openmw/pull/2047). I would have looked at it when it was first made and seen a `recreateShaders` call which, at the time, matched the surrounding code, so didn't warrant any comment. For whatever reason, this hung around on a branch until late 2019, and it was rebased several times. One of those pulled in bzzt's change making the `recreateShaders` call unnecessary, but it seems this was missed, and the call wasn't removed. Eventually, it ended up on the master branch as it was missed in code review, and then as it's the last vestige of an old implementation, stuck out and looked like a weird workaround.

Reply:
I did not think it was worth figuring out and I am not going to bother to read this message, but that is still good news I suppose.